### ISSUE TITLE
FreeRDP not handling properly connection where there are no more licenses available

### ISSUE THREAD

**Issue by bogdancondurache** [FreeRDP/FreeRDP#8107]
Hello all!

We're currently facing issues with FreeRDP when connecting to a server that has already reached its maximum number of connections based on licenses. 

The first problem is that FreeRDP will display the UI for kicking someone out for an admin user, but for non-admin user it will just have a black screen. We would really like to be able to handle this case, for example Microsoft RDP Client returns an error when this happens.

The second issue is that sometimes when this happens, FreeRDP is not properly closing the session, leaving them idle for a very long time.

This seems to only be happening on Windows, the system we used for testing is x64, latest version of FreeRDP.
One way we managed to reproduce this is by having 2 users connect at the same time to eat up both licenses and then have a non-admin user also try to connect.

**Comment by hardening** [comment]
Can you send us some logs with debug activated (`/log-level:DEBUG`) so that we can see where the connection gets stuck (black screen). I'm suspecting that it happens during licence negotiation and that we misinterpret the returned error code, but I want to confirm that.

**Comment by Sy55** [comment]
Ok

**Comment by bogdancondurache** [comment]
Hello, @hardening! Unfortunately it looks like we have 2 separate issues. On the machine we thought we managed to reproduce it, it was actually because of low resources. It still seems bug-ish, because the process simply freezes because of not enough memory, without reporting anything and the Microsoft RDP client works without a problem on that case. So we thought we reproduced first issue, but it might be a different one. So at the moment we can't really reproduce the initial issue, but if you think the out of memory one is also an issue, I can open another one for it.

Also, on Windows client it seems that logs (ie. `/log-level:DEBUG`) are not shown on STDOUT and the output has to be redirected somewhere or else I won't see any output at all
